design/Proposal-to-enhance-schema-for-metal3.md

```diff
@@ -0,0 +1,228 @@
+
+Please refer `metal3` spec for bare-metal:
+https://github.com/metal3-io/baremetal-operator/blob/master/deploy/crds/metal3.io_baremetalhosts_crd.yaml
+
+The investigation of the existing schema indicates that BareMetal-Operator 
+gets BareMetal node introspection data using an API call to `Ironic-Inspector` 
+through `gopher-cloud client` library. 
+
+Having fetched BareMetal Node introspection data from Ironic-Inspector and
+compared it with BareMetalHost schema of `metal3` `(metal3.io_baremetalhosts_crd.yaml`), 
+we recommend following parameters need to be added in schema of `metal3` for `NFV` 
+workloads.
+1. Add enhancement in the existing `metal3` schema for the following parameters:
+
+    CPU Parameters:
+    
+    Parameters for `NUMA` topology:
+     
+    For better performance, selection of CPUs based on the `NUMA` topology 
+    becomes necessary. In case of nodes with `DPDK` aware NICs, the CPUs for 
+    poll mode driver (PMD) need to be selected from the `NUMA` node associated 
+    with the `DPDK` NICs. If hyperthreading is enabled, then selection of the 
+    logical cores requires the knowledge of the siblings.  
+
+    Declarative design idea:
+    ````yaml
+    Cpu
+     …
+    numa_topology:
+      properties:
+        numa_node_0:
+          properties:
+            nics:
+              properties:
+                numa_node:
+                  type: integer
+                  description: Numa node id
+                name:
+                  type: string
+                  description: network interface name
+              type: object
+            type: array
+            ram:
+              properties:
+                numa_node:
+                  type: integer
+                  description: Numa node id
+                size_kb:
+                  type: string
+                  description: memory in kb
+              type: object
+            type: array
+            cpus:
+              properties:
+                cpu:
+                  type: integer
+                  description: cpu id 
+                numa_node:
+                  type: integer
+                  description: Numa node id
+                thread_siblings:
+                  items:
+                  type: integer
+                type: array
+                description: list of thread_siblings for each core
+              type: object
+            type: array
+        …
+      required:
+      - nics
+      - ram
+      - cpu
+      type: object
+    ````
+    
+    Where,
+    * `ram` a mapping from memory available to a `NUMA` node
+    * `cpus` a mapping from physical CPU ID to a `NUMA` node and a list of sibling threads
+    * `nics` a mapping from NIC names to `NUMA` node
+
+    Boot Parameters:
+    
+    * current_boot_mode: `uefi`
+        
+        UEFI is installed when device is manufactured. It is the first program which runs when the device is turned on. UEFI replace BIOS.
+    * pxe_interface: `ff:ff:ff:ff:ff:ff`
```

Comment:
How is this different from [the existing bootMACAddress field](https://github.com/metal3-io/baremetal-operator/blob/master/pkg/apis/metal3/v1alpha1/baremetalhost_types.go#L129)?

Reply:
Its same but it makes sense if we keep it under boot mode section because that way, user get to know which pxe nic is associated with boot mode.

Reply:
The bootMACAddress *is* the MAC we expect to be used for PXE.

Reply:
Anyway I am creating seperate PR For boot. will address these things there.